test/auth/features/oidc/broker.go

```diff
@@ -101,22 +113,35 @@ func BrokerSendEventWithOIDCTokenToDLS() *feature.Feature {
 	// Install DLS sink
 	f.Setup("install dead letter sink", eventshub.Install(dls,
 		eventshub.OIDCReceiverAudience(dlsAudience),
-		eventshub.StartReceiver))
-
-	// Install broker with DLS config
-	brokerConfig := append(
-		broker.WithEnvConfig(),
-		delivery.WithDeadLetterSinkFromDestination(&duckv1.Destination{
-			Ref:      service.AsKReference(dls),
-			Audience: &dlsAudience,
-		}),
-	)
-	f.Setup("install broker", broker.Install(brokerName, brokerConfig...))
+		eventshub.StartReceiverTLS))
+
+	f.Setup("install broker", func(ctx context.Context, t feature.T) {
+		brokerConfig := append(broker.WithEnvConfig(),
+			delivery.WithDeadLetterSinkFromDestination(&duckv1.Destination{
+				Ref:      service.AsKReference(dls),
+				Audience: &dlsAudience,
+				//CACerts: eventshub.GetCaCerts(ctx),
+			}))
+		broker.Install(brokerName, brokerConfig...)(ctx, t)
+	})
+
 	f.Setup("Broker is ready", broker.IsReady(brokerName))
 
 	// Install Trigger
-	f.Setup("install trigger", trigger.Install(triggerName, brokerName,
-		trigger.WithSubscriber(nil, "bad://uri")))
+	//f.Setup("install trigger", trigger.Install(triggerName, brokerName,
+	//	trigger.WithSubscriber(nil, "bad://uri")))
+
+	f.Setup("install the trigger and specify the CA cert of the destination", func(ctx context.Context, t feature.T) {
+		// create an empty destination ref
+		d := service.AsDestinationRef("")
+		d.CACerts = eventshub.GetCaCerts(ctx)
+		// uri is an addressable, create a new one and put the bad	uri in it
+		d.URI, _ = apis.ParseURL("bad://uri")
+		trigger.Install(triggerName, brokerName, trigger.WithSubscriberFromDestination(d))(ctx, t)
+
```

Comment:
event -> broker -> trigger -> subscriber ->because subscriber's url is not valid -> DLS

And during this process, the communication between trigger and subscriber, trigger and DLS should be TLS encrypted, is that correct? @creydr

Reply:
yes